Added actionOpenClosed for contactSensors (simulated)

iprak opened the pull request:
Allow open/close to be conditionally invoked on simulated contact sensors. [Contact sensors](http://docs.smartthings.com/en/latest/capabilities-reference.html#contact-sensor)  do not have open/close command. Simulated contact sensor with no actionable tile would be one way to update a SmartThings contact sensor through MQTT.

iprak committed:
Added actionOpenClosed for contactSensors (simulated)

iprak committed:
Don't add duplicate subscription

iprak committed:
Remove unused variable

gandazgul commented on smartapps/stj/mqtt-bridge.src/mqtt-bridge.groovy:
```diff
@@ -611,11 +612,17 @@ def actionColor(device, attribute, value) {
 def actionOpenClosed(device, attribute, value) {
     if (value == "open") {
-        device.open()
+		if (device.hasCommand("open")) {
```
Please use spaces for indentation

gandazgul commented on smartapps/stj/mqtt-bridge.src/mqtt-bridge.groovy:
```diff
@@ -721,4 +728,4 @@ def actionTimedSession(device, attribute, value) {
         device.setTimeRemaining(value)
     }
-}
+}
```
Please make sure there's a new line at the end of the file

gandazgul commented on server.js:
```diff
@@ -195,8 +195,12 @@ function handleSubscribeEvent (req, res) {
+            var cmdTopic = getTopicFor(device, property, TOPIC_COMMAND);
+            var writeTopic = getTopicFor(device, property, TOPIC_WRITE_STATE);
+            subscriptions.push(cmdTopic);
+            if (writeTopic !== cmdTopic) {
```
nice catch!

iprak committed:
Fixed white spaces

gandazgul commented:
@stjohnjohnson this would be a nice one to merge.

iprak committed:
Only setting command if piece[2] is present and matches commandSuffix

iprak committed:
Removed garageDoors entry since it has been deprecated in favor of doorControl.
Added "opensensor", "closedsensor", "switch", "notify" to doorControl
Allowing "notify" to be sent on devices.

iprak committed:
Adjusted logging

iprak committed:
Remove "switch" custom attribute from "Door Control", invoke action as fallback if setStatus is absent

BrettSheleski commented:
Please merge this.

iprak committed:
Removed capabilities for which I don't have any devices
Changed door to door, notify to match EspurnaGarageDoorDeviceHandlerV2
Force sent device battery status when initialized

iprak committed:
Retry if connection fails
